src/lib/profiles/data-management/Current/CommandSender.h

```diff
@@ -0,0 +1,261 @@
+/*
+ *
+ *    Copyright (c) 2016-2017 Nest Labs, Inc.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines command handle for Weave
+ *      Data Management (WDM) profile.
+ *
+ */
+
+#ifndef _WEAVE_DATA_MANAGEMENT_COMMAND_SENDER_CURRENT_H
+#define _WEAVE_DATA_MANAGEMENT_COMMAND_SENDER_CURRENT_H
+
+#include <Weave/Profiles/data-management/Current/WdmManagedNamespace.h>
+#include <Weave/Core/WeaveCore.h>
+#include <Weave/Profiles/data-management/Current/TraitCatalog.h>
+#include <Weave/Profiles/data-management/Current/Command.h>
+
+namespace nl {
+namespace Weave {
+namespace Profiles {
+namespace WeaveMakeManagedNamespaceIdentifier(DataManagement, kWeaveManagedNamespaceDesignation_Current) {
+
+/**
+ *  @class CommandSender
+ *
+ *  @note This class encapsulates the protocol details of sending commands, simplifying the mechanics involved for
+ *        applications. The application provides a PacketBuffer containing the payload of the command as well as
+ *        an optional set of arguments that alter the contents of the command header as well as the behavior of the command.
+ *
+ *        The utility of this wrapper around command handling is indeed limited, mainly due to the complexity/flexibility involved
+ *        in security validation and data serialization/de-serialization.
+ *
+ *        The details for command validation is still TBD
+ *
+ *        This class also helps applications infer if the data within an associated TraitDataSink has caught up to the side-effects
+ *        of the commmand (based on the version provided in the command response). The application is responsible for managing the
+ *        storage of that object.
+ *
+ * ## Weave Binding
+ *
+ * An object of this class can be intialized with a Weave Binding, which will serve as the default
+ * Binding to use to send Commands. The user may also provide a Binding to each call of
+ * SendCommand(), which will override the default Binding. It is not necessary to provide a default
+ * Binding, however any binding provided to CommandSender must already be initialized.
```

Comment:
@mrjerryjohns @robszewczyk Is it acceptable to require that the user has already initialized the binding?